torch/nn/attention/flex_attention.py

```diff
@@ -1132,9 +1132,13 @@ def _validate_device(query: Tensor, key: Tensor, value: Tensor):
     """TODO: Remove once non cuda/cpu devices support is added
     We only need to check query since we have already that q,k,v are on the same device
     """
-    if query.device.type != "cuda" and query.device.type != "cpu":
+    if (
+        query.device.type != "cuda"
+        and query.device.type != "cpu"
+        and query.device.type != "hpu"
+    ):
         raise ValueError(
-            "FlexAttention is only supported on CUDA or CPU devices. "
+            "FlexAttention is only supported on CUDA, CPU or HPU devices. "
```

Comment:
Does Ascend NPU support this? If so, we should relax this restriction to support privateuse1 @FFFrog

Reply:
A small note, HPU has a dedicated dispatch key. So privateuse1 should be addition to it.

Reply:
Yes, should support PrivateUse1 for out-of-tree backend